### ISSUE TITLE
Exported path doesn't work when a dependency of Atom Packages

### ISSUE THREAD

**Issue by francoislaberge** [electron-userland/electron-prebuilt#99]
Atom installs packages to a temporary location before copying it to another if a package successfully installs. This makes the path.txt file's content incorrect as it writes out the original installation location. Our workaround was implementing the following function and using it instead of the exported path of the ```electron-prebuilt``` module. This doesn't look much more expensive than [the current implementation](https://github.com/electron-userland/electron-prebuilt/blob/1d6103f0579b766ee49977b618dd04a2b0bc8fe1/index.js#L1-L4) and either way, this isn't code that is going to be called repeatedly in any performance critical code.

```js
/**
 * Returns where the electron executable is
 *
 * We have to reimplement this versus just using 'const path = require('electron-prebuilt');' because
 * when jibo-tools is installed via Atom's UI or 'apm install' all modules are installed temporarily in a temp
 * directory and then moved somewhere else if successful, this breaks electron-prebuilts postinstall logic
 * that saves out a file that includes the path to the electron executable
 */
function getElectronPath() {
    var path = require("path"),
        os = require("os"),
        platform = os.platform(),
        paths = {
            darwin: "dist/Electron.app/Contents/MacOS/Electron",
            freebsd: "dist/electron",
            linux: "dist/electron",
            win32: "dist/electron.exe"
        },
        // Hoping this makes it NPM3 compatible also
        electronPath = path.resolve( path.dirname(require.resolve('electron-prebuilt')), paths[platform]);

    return electronPath;
}
```

**Comment by maxogden** [comment]
Interesting, I've never used atom packages, does it let you install from npm? (Just curious how you are using this from inside Atom)

**Comment by francoislaberge** [comment]
You can install via their ```apm``` command line wrapper around npm (it has to replace node-gyp to compile native code against Atom's headers/libraries) and this is also used internally when you use the UI to install packages from the Atom package registry. 

You can just use ```npm install``` if your code doesn't have native code though.

**Comment by maxogden** [comment]
Ah I see, so I'm guessing for some reason you want to support apm and cant use npm instead?

If you make this into a PR you should just put this https://github.com/electron-userland/electron-prebuilt/blob/master/install.js#L19-L30 into a separate .js file that both index.js and install.js use to DRY it up.

**Comment by francoislaberge** [comment]
In order for any Atom Package to include electron-prebuilt as a dependency, direct or not, we need this change for the exported path to be correct. In our case we want our [Atom pacakge](https://atom.io/packages/jibo-sdk) to include it's version of electron-prebuilt, so we then run into this issue. Anyone using electron-prebuilt as a dependency will face this problem. I think it's a Mac only issue, but I may be remembering incorrectly.

Sure will do, about to jump on a flight to SF, will try to get to this this tomorrow.

**Comment by mafintosh** [comment]
@francoislaberge i don't understand the difference between what your example does and what we currently do in index.js. when wouldn't path.join(__dirname, relativePathToDist) not work?

**Comment by francoislaberge** [comment]
My bad guys. We're stuck at 0.34.1 for reasons that don't matter in regard to this. Looks like you changed things to be relative since then. It was fixed December 10th in [this commit](https://github.com/electron-userland/electron-prebuilt/commit/1d6103f0579b766ee49977b618dd04a2b0bc8fe1) and released with [0.36.9](https://github.com/electron-userland/electron-prebuilt/releases/tag/v0.36.9).

Before:
```js
var paths = {
  darwin: path.join(__dirname, './dist/Electron.app/Contents/MacOS/Electron'),
  linux: path.join(__dirname, './dist/electron'),
  win32: path.join(__dirname, './dist/electron.exe')
}
```

Now:
```js
var paths = {
  darwin: 'dist/Electron.app/Contents/MacOS/Electron',
  freebsd: 'dist/electron',
  linux: 'dist/electron',
  win32: 'dist/electron.exe'
}
```

Closing this. This issue will go away when we upgrade.
